### ISSUE TITLE
Add Card holder name next to obfuscated card number

### ISSUE THREAD

**Issue by codrut-topliceanu** [mozilla-mobile/fenix#19961]
[It's been suggested](https://github.com/mozilla-mobile/fenix/pull/19687#issuecomment-857829776) that we should display the card holder's name next to the obfuscated credit card number:

> > > @codrut-topliceanu I checked with @topotropic and we want to add the name on card instead of network names to align with desktop. Sorry!
> > 
> > 
> > I just realized you already said that in the issue and you were referring to the card holder name, not the `card.name`.
> > Should I close the PR and issue? Or change the code here, while keeping the issue? What is the recommended procedure here?
> 
> I think creating a new PR probably makes sense to preserve this work in case we ever want to come back to it. Having said that, I think it would be best to just file a new issue as well to preserve the history of your changes and the intentions behind. If we decided to come back to the network names, then we can just look up this PR in the future.

I'm not entirely sure how we should go about this as it's very likely that the credit cards on a user's device will all have his name on them, right?

<img src="https://user-images.githubusercontent.com/60002907/121639643-85e13980-ca95-11eb-9211-7262229a0d3d.png" width=300 />

Looking for further input on this from UX.

**Comment by topotropic** [comment]
> > 
> 
> I'm not entirely sure how we should go about this as it's very likely that the credit cards on a user's device will all have his name on them, right?

Not sure if I understand your questions, can you rephrase it? Thank you!

**Comment by codrut-topliceanu** [comment]
> > > 
> > 
> > 
> > I'm not entirely sure how we should go about this as it's very likely that the credit cards on a user's device will all have his name on them, right?
> 
> Not sure if I understand your questions, can you rephrase it? Thank you!


All my credit cards have my name written on them 🤷‍♂️ . So the question would be: is it useful/desirable for me to see my name next to the obfuscated number on the credit cards that i've saved in Fenix?

**Comment by topotropic** [comment]
Thanks! That's a good point. So it might be better to add something that helps people differentiate it – maybe the obfuscated number is enough (?!). But we would still store the name just not show it, correct?

**Comment by codrut-topliceanu** [comment]
@topotropic Yep, that's correct. We do store the name on the card.

At the moment to differentiate between saved credit cards the user has:

- last 4 digits of the credit card number
- credit card network icon

I've seen that other apps also display the credit card network name (Mastercard, Visa, etc), sometimes even the expiry date of the card on a 2nd row of text, beneath the last 4 digits row. Not sure if that last part is a good idea though.

I suppose we could add a new field to the credit card object to "name" a card. A human readable identifier chosen by the user. For example I have a credit card called "small payments" and another called "online payments" (but this is for a very specific banking app, I haven't seen it done anywhere else). This route would require a bit of changes on AC and Fenix though.
